code/core/write_sdl_ntuple.cc

```diff
@@ -199,114 +198,46 @@ std::tuple<float, float, float, vector<unsigned int>, vector<unsigned int>> pars
-    unsigned int T5_md1 = segmentsInGPU.mdIndices[2 * LS1];
-    unsigned int T5_md2 = segmentsInGPU.mdIndices[2 * LS2];
-    unsigned int T5_md3 = segmentsInGPU.mdIndices[2 * LS3];
-    unsigned int T5_md4 = segmentsInGPU.mdIndices[2 * LS4];
-    unsigned int T5_md5 = segmentsInGPU.mdIndices[2 * LS4 + 1];
-
-    // Getting Hit indices
-    unsigned int pLS_hit1 = miniDoubletsInGPU.anchorHitIndices[pLS_md1];
-    unsigned int pLS_hit2 = miniDoubletsInGPU.outerHitIndices [pLS_md1];
-    unsigned int pLS_hit3 = miniDoubletsInGPU.anchorHitIndices[pLS_md2];
-    unsigned int pLS_hit4 = miniDoubletsInGPU.outerHitIndices [pLS_md2];
-    unsigned int T5_hit1  = miniDoubletsInGPU.anchorHitIndices[T5_md1];
-    unsigned int T5_hit2  = miniDoubletsInGPU.outerHitIndices [T5_md1];
-    unsigned int T5_hit3  = miniDoubletsInGPU.anchorHitIndices[T5_md2];
-    unsigned int T5_hit4  = miniDoubletsInGPU.outerHitIndices [T5_md2];
-    unsigned int T5_hit5  = miniDoubletsInGPU.anchorHitIndices[T5_md3];
-    unsigned int T5_hit6  = miniDoubletsInGPU.outerHitIndices [T5_md3];
-    unsigned int T5_hit7  = miniDoubletsInGPU.anchorHitIndices[T5_md4];
-    unsigned int T5_hit8  = miniDoubletsInGPU.outerHitIndices [T5_md4];
-    unsigned int T5_hit9  = miniDoubletsInGPU.anchorHitIndices[T5_md5];
-    unsigned int T5_hit10 = miniDoubletsInGPU.outerHitIndices [T5_md5];
 
     // pLS  1    2    3    4    5
     // **** o -- o -- o -- o -- o
     //      o -- o -- o
     //                o -- o -- o
-    unsigned int i1_idx = T5_hit1;
-    unsigned int i3_idx = T5_hit5;
-    unsigned int i5_idx = T5_hit9;
-
-    const float dr_in = sqrt(pow(hitsInGPU.xs[i3_idx] - hitsInGPU.xs[i1_idx], 2) + pow(hitsInGPU.ys[i3_idx] - hitsInGPU.ys[i1_idx], 2));
-    const float dr_out = sqrt(pow(hitsInGPU.xs[i5_idx] - hitsInGPU.xs[i3_idx], 2) + pow(hitsInGPU.ys[i5_idx] - hitsInGPU.ys[i3_idx], 2));
-    float betaIn_in   = __H2F(tripletsInGPU.betaIn[T3Idx1]);
-    float betaOut_in  = __H2F(tripletsInGPU.betaOut[T3Idx1]);
-    float betaIn_out  = __H2F(tripletsInGPU.betaIn[T3Idx2]);
-    float betaOut_out = __H2F(tripletsInGPU.betaOut[T3Idx2]);
+    unsigned int pT5 = trackCandidatesInGPU.directObjectIndices[idx];
+    std::vector<unsigned int> Hits = getHitsFrompT5(event, pT5);
+    unsigned int Hit_0 = Hits[0];
+    unsigned int Hit_4 = Hits[4];
+    unsigned int Hit_8 = Hits[8];
+
+    std::vector<unsigned int> T3s = getT3sFrompT5(event, pT5);
+    unsigned int T3_0 = T3s[0];
+    unsigned int T3_1 = T3s[1];
+
+    unsigned int pLS = getPixelLSFrompT5(event, pT5);
+
     const float kRinv1GeVf = (2.99792458e-3 * 3.8);
     const float k2Rinv1GeVf = kRinv1GeVf / 2.;
+    const float dr_in = sqrt(pow(hitsInGPU.xs[Hit_4] - hitsInGPU.xs[Hit_0], 2) + pow(hitsInGPU.ys[Hit_4] - hitsInGPU.ys[Hit_0], 2));
+    const float dr_out = sqrt(pow(hitsInGPU.xs[Hit_8] - hitsInGPU.xs[Hit_4], 2) + pow(hitsInGPU.ys[Hit_8] - hitsInGPU.ys[Hit_4], 2));
```

Comment:
Related to the comment above, what are we computing here exactly? If I understand correctly, these hits are not on the same segment, or am I mistaken?

In any case, I think this shows that there are still some points that are not clear in the code. And I know this change is not introduced by this PR and it is an opportunity for others to clarify what we are doing.

Reply:
I added some more information... with some history

Reply:
Thanks for the summary, it was useful.

In any case, my questions were originating from the meaning of these variables before the latest bug fix. After that, I think that everything makes sense.